test/lint/lint-all.sh

```diff
@@ -16,21 +16,19 @@ set -u
 SCRIPTDIR=$(dirname "${BASH_SOURCE[0]}")
 LINTALL=$(basename "${BASH_SOURCE[0]}")
 
-for f in "${SCRIPTDIR}"/lint-*.sh; do
-  if [ "$(basename "$f")" != "$LINTALL" ]; then
-    if ! "$f"; then
-      echo "^---- failure generated from $f"
-      exit 1
-    fi
-  fi
-done
+# TODO: Replace with "${SCRIPTDIR}"/lint-*.sh
+SHELL_SCRIPTS=(
+  "${SCRIPTDIR}"/lint-circular-dependencies.sh
+  "${SCRIPTDIR}"/lint-filenames.sh
+  "${SCRIPTDIR}"/lint-python-shebang.sh
+  "${SCRIPTDIR}"/lint-whitespace.sh
+)
 
-for f in "${SCRIPTDIR}"/lint-*.py; do
```

Comment:
This disables the python lint scripts. We should reenable them as well. The todo above is a bit misleading then, because it only refers to the shell scripts.

Reply:
There's only one Python script, `lint-format-strings.py`, and it has a shell script wrapper, so this section can be removed without issues.

Reply:
...unless we repurpose this section to run the newly added `check-*.py` scripts, maybe?

Reply:
I actually like your approach of specifying the scripts to run explicitly. Then it's clear what is run and in which sequence, and it's also easy to disable a script. So maybe we can just keep that instead of going back to the directory grepping magic. Then it also doesn't matter if it's shell or python or anything else.